src/application/downloader.py

```
@@ -0,0 +1,213 @@
+    async def _download_location_with_limit(
+        self,
+        location: Location,
+        job: DownloadJob,
+        index: int,
+        total: int,
+        semaphore: asyncio.Semaphore
+    ) -> None:
+        async with semaphore:
+            await self._download_location(location, job, index, total)
+
+    @MetricsMiddleware.track_download("location")
```

Pylint (3.3.7)

[error] No value for argument 'source' in unbound method call

(E1120)